app/controllers/api/v1x0/orders_controller.rb

```diff
@@ -14,6 +14,8 @@ def create
       def submit_order
         approval = Catalog::CreateApprovalRequest.new(params.require(:order_id))
         render :json => approval.process.order
+      rescue Catalog::ApprovalError => e
+        render :json => { :errors => e.message }, :status => :internal_server_error
```

Comment:
I'm not understanding the 500 here. A 500 would say that Approval had an internal 'explosion' but in our case a user didn't assign a value. I think the 500 will come across as something is down.

Reply:
I was thinking the same as @syncrou here.  I think of the 500 as a coding or system error, not a misconfiguration.

> 422 Unprocessable Entity response status code indicates that the server understands the content type of the request entity, and the syntax of the request entity is correct, but it was unable to process the contained instructions.

> 500 Internal Server Error server error response code indicates that the server encountered an unexpected condition that prevented it from fulfilling the request.

Reply:
@syncrou @gmcculloug 
If this is a data error, then we should have caught it when the record was being created for the first time. But since we have the internal logic that says we are going to allow for the approval to come from either the platform, project or portfolio, the end user doesn't know that we allowed him to create a Portfolio with a blank value and then complained when he went to order. 4xx errors imply that there is something wrong with the what the user entered.

Reply:
Assume we were treating this like an enable/disable flag and that the UI could respond properly and enable/disable the order button.

A user could still try to order through the API regardless of what the property was set to.  What is the correct HTTP return code in that scenario?

Reply:
@gmcculloug 
Is this enable/disable a user settable attribute or a runtime determination?

Would it be better if we always require the Portfolio to have a approval workflow and have it be overridden a the Platform, Project level. The user can set Auto Approve by default but the Admin could set the Approval at the Platform/Project level to go thru extra approval.

Reply:
Maybe 406? https://httpstatuses.com/406